rewrite-java-test/src/test/java/org/openrewrite/java/recipes/SourceSpecTextBlockIndentationTest.java

```diff
@@ -83,4 +83,56 @@ void test() {
           )
         );
     }
+
+    @Test
+    void startsOnNewline() {
+        rewriteRun(
+          java(
+            """
+              import org.openrewrite.test.RewriteTest;
+              import static org.openrewrite.test.SourceSpecs.text;
+
+              class MyRecipeTest implements RewriteTest {
+                  void test() {
+                    rewriteRun(
+                       text(\"""
+                           class Test {
+              \s
+                              \s
+                               void test() {
+                                   System.out.println("Hello, world!");
+                               }
+                           }
+                           \"""
```

Comment:
Possibly we could also change the recipe to put the closing parenthesis following the text block on its own line. Quite often I see it be on the same line.

Reply:
Good addition indeed; I've also thought that maybe we should detect placement of text blocks on a new line as a style option, and aim for consistency that way rather than with this addition here. Might be more involved though.

Reply:
I'm going to merge this one as is, as I don't immediately see an easy way to add a suffix to the text block under the right conditions, and as implemented now it's already an improvement that might prompt a rethink of newlines on contributions. Thanks for the suggestion; can be picked up separately.